doc/src/vpicUnits.tex

```diff
@@ -37,41 +51,156 @@
+
+	\noindent where $\ncrit$ is the critical density $\ncrit \equiv m_e\varepsilon_0\omega_0^2/e^2$ In this unit system the simulation cell size and time step should satisfy $\Delta x \ll 1$ and $\Delta t \ll 1$ in order for the field solver to accurately model the laser. It is also necessary to verify that the cell size resolves the Debye length, which is often a more restrictive condition. Additionally, the normalised electric (or magnetic) field is the parameter $a_0$. For $a_0 \ll 1$ this is the electron quiver velocity divided by the speed of light, while $a_0 \gtrsim 1$ indicates a relativistic field intensity which will lead to effects such as relativistically induced transparency and relativistic self-focusing.
+
+	\section{Species \& Particle Weighting}
+
+	To define a particle species in VPIC, the user specifies the species mass $m_{\mathrm{s}}$ and charge $q_{\mathrm{s}}$. Additionally, each particle is individually assigned a weighting factor $w$. These must be consistent with each other and the above unit system to produce the desired behaviour.
+
+	\subsection{Choosing Charge Mass Ratio}
+
+	The motion of a physical particle with charge $q_{\mathrm{p}}$ and mass $m_{\mathrm{p}}$ in the EM fields only has explicit dependence on the charge-mass ratio $r \equiv q_{\mathrm{p}}/m_{\mathrm{p}}$. This is also the case for macroparticles in VPIC's particle push. Therefore the corresponding macroparticle species must be defined such that its charge $q_{\mathrm{s}}$ and mass $m_{\mathrm{s}}$ satisfy $q_{\mathrm{s}}/m_{\mathrm{s}}=r$.
+
+	\subsection{Choosing Macroparticle Weight}
+	When calculating the fields generated by a macroparticle, the code uses the species charge and macroparticle weight to calculate a total charge for the macroparticle:
 
-	Particle momenta are the only user-facing exception to the unit system defined above. The momenta are normalised to produce a dimensionless momentum
-	
 	\begin{equation}
-		u \equiv \frac{p}{mc}.
+		q_{\mathrm{MP}} = wq_{\mathrm{s}}.
 	\end{equation}
+
+	\noindent For this to result in an amount of current deposition consistent with the desired physical particle density, the macroparticle weight should be defined such that the charge contained in volume $V$ is the same for the macroparticles as for physical particles, i.e.
+
+	\begin{align*}
+		q_{\mathrm{p}}n_{\mathrm{p}}V &= q_{\mathrm{MP}}n_{\mathrm{MP}}V \\
+			&= wq_{\mathrm{s}}n_{\mathrm{MP}}V,
+	\end{align*}
+
+	\noindent where $n_{\mathrm{p}}$ and $n_{\mathrm{MP}}$ are the physical particle and macroparticle number densities (macroparticles per unit volume) respectively. So, the particle weight must satisfy
+
+	\begin{align}
+		w = \frac{q_{\mathrm{p}}n_{\mathrm{p}}}{q_{\mathrm{s}}n_{\mathrm{MP}}} = \frac{\rho_{\mathrm{p}}}{\rho_{\mathrm{s}}}.
+	\end{align}
+
+	\noindent This illustrates that there is a degree of freedom available in that the species charge defined in the deck does not have to be the same as the physical species charge.
```

Comment:
I think this is really the key insight to understanding units in VPIC.  The physical properties of particles are independent of the equations VPIC solves, so there are many ways you can coax the math into representing your physical system.  I'd like to see this as a running theme throughout.

Reply:
Could you suggest more specifically what should be changed/added?

My understanding is that the user can set various constants that determine the unit system they use for modeling their problem, and on top of that has the freedom to choose an arbitrary macroparticle species charge provided that the particle weight is consistent with the physical particle charge density. Have I missed something?